casanovo/denovo/model_runner.py

```
@@ -291,29 +296,86 @@ def predict(
+        if profile_output is not None:
+            profiler_ctx = torch.profiler.profile(
+                activities=[ProfilerActivity.CPU, ProfilerActivity.CUDA],
+                record_shapes=True,
+                profile_memory=True,
+                with_stack=False,
+            )
+        else:
+            profiler_ctx = contextlib.nullcontext()
+
+        with utils.stage_timer("inference", stage_times):
+            with profiler_ctx as prof:
```

Ruff (0.15.10)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

Combine `with` statements

(SIM117)